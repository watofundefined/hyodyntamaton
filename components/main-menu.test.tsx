import React from 'react'
import { render, fireEvent } from 'test/test-utils'
import MainMenu from './main-menu'

const options: [string, string][] = [
  ['Pubs Nearby', '/map'],
  ['Account', '/account'],
  ['About', '/about'],
]

describe('MainMenu component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<MainMenu nav={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders button for each nav item', () => {
    const { getAllByRole } = render(<MainMenu nav={options} />)
    const buttons = getAllByRole('button')
    expect(buttons).toHaveLength(options.length)
  })

  it('redirects after button is clicked', () => {
    const push = jest.fn()
    const { getByText } = render(<MainMenu nav={options} />, { router: { push } })
    const [name, route] = options[0]
    const button = getByText(new RegExp(name, 'i'))

    fireEvent.click(button)

    expect(push).toHaveBeenCalledTimes(1)
    expect(push).toHaveBeenCalledWith(route)
  })
})
