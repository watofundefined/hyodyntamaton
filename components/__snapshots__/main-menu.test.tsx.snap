// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component matches snapshot 1`] = `
<DocumentFragment>
  <button
    __self="[object Object]"
    __source="[object Object]"
    class="btn"
  >
    Pubs Nearby
  </button>
  <button
    __self="[object Object]"
    __source="[object Object]"
    class="btn"
  >
    Account
  </button>
  <button
    __self="[object Object]"
    __source="[object Object]"
    class="btn"
  >
    About
  </button>
</DocumentFragment>
`;
